src/Components/Gifs/Item.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { connect } from "react-redux";
+import { ItemStyled, ItemButtonStyled } from "./Item.styled";
+
+const Item = ({
+  item,
+  favorites: { items: favoriteItems },
+  onToggleFavorite
+}) => {
+  const {
+    title,
+    images: {
+      fixed_width: { url, height, width }
+    }
+  } = item;
+
+  return (
+    <ItemStyled>
+      <img src={url} alt={title} width={width} height={height} />
+      <ItemButtonStyled
+        onClick={() => {
+          onToggleFavorite(item);
+        }}
+        isFavorite={item.id in favoriteItems}
+      >
+        <i className="far fa-thumbs-up" />
+      </ItemButtonStyled>
+    </ItemStyled>
+  );
+};
+
+const mapStateToProps = state => {
+  const { favorites } = state;
+  return {
+    favorites
+  };
+};
+
+const mapDispatchToProps = dispatch => {
+  return {
+    onToggleFavorite: item =>
+      dispatch({ type: "FAVORITE_TOGGLE", payload: { item } })
```

Comment:
You can use `bindActionCreators` or the object notation to avoid having to wrap the dispatch manually around the creators.